File: MeowCore/Models/RequestDTOs/TodoRequestDto.cs
using System.ComponentModel.DataAnnotations;
using MeowCore.Helpers;

namespace MeowCore.Models.RequestDTOs
{
    public class TodoRequestDto
    {
        [Required]
        [MaxLength(250)]
        public string Title { get; set; } = string.Empty;

        public string? Description { get; set; }

        public TodoStatus Status { get; set; } = TodoStatus.Pending;

        [Required]
        public int ListId { get; set; }
    }
}

File: MeowUIX/src/services/Interfaces/TodoRequestDto.ts
export type TodoStatus = 0 | 1 | 2 | 3; {/* 0: Pending, 1: InProgress, 2: Completed, 3: Archived */}

export interface TodoRequestDto {
    title: string;
    description?: string;
    listId: number;
    status?: TodoStatus;
}